internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -709,16 +608,27 @@ func resourceRecoveryServicesVaultRead(d *pluginsdk.ResourceData, meta interface
 		d.Set("sku", string(sku.Name))
 	}
 
-	if model.Properties != nil && model.Properties.SecuritySettings != nil && model.Properties.SecuritySettings.ImmutabilitySettings != nil {
-		d.Set("immutability", string(pointer.From(model.Properties.SecuritySettings.ImmutabilitySettings.State)))
-	}
+	if prop := model.Properties; prop != nil {
+		if prop.SecuritySettings != nil && prop.SecuritySettings.ImmutabilitySettings != nil {
+			d.Set("immutability", string(pointer.From(prop.SecuritySettings.ImmutabilitySettings.State)))
+		}
 
-	if model.Properties != nil && model.Properties.PublicNetworkAccess != nil {
-		d.Set("public_network_access_enabled", flattenRecoveryServicesVaultPublicNetworkAccess(model.Properties.PublicNetworkAccess))
-	}
+		if prop.PublicNetworkAccess != nil {
+			d.Set("public_network_access_enabled", flattenRecoveryServicesVaultPublicNetworkAccess(model.Properties.PublicNetworkAccess))
+		}
+
+		if prop.MonitoringSettings != nil {
+			d.Set("monitoring", flattenRecoveryServicesVaultMonitorSettings(*model.Properties.MonitoringSettings))
+		}
```

Comment:
The Flatten function should handle this value being nil and set an appropriate default value (e.g. false) else this is going to be `null` meaning users won't be able to interpolate the value